### ISSUE TITLE
Concurrent read and write to FileReaders - Web Assembly

### ISSUE THREAD

**Issue by semvis123** [klippa-app/go-pdfium#73]
Using file readers in `OpenDocument` is not thread safe.
It causes a concurrent read and write to map.

This is the map that is being accessed from multiple places at once. 
`var FileReaders = map[uint32]*FileReaderRef{}`

Expected behavior:
OpenDocument working with Reader as input in a multithreaded WASM configuration.

Current workaround:
Read the file into memory and pass it as a byte array.

Possible solution:
`sync.Mutex` for accessing this map, or the usage of `sync.Map`
`sync.Mutex` is probably the best option here, because it keeps the type safety.

Error logs:
```
fatal error: concurrent map read and map write

goroutine 77 [running]:
github.com/klippa-app/go-pdfium/webassembly/imports.FPDF_FILEACCESS_CB.Call({}, {0x2000?, 0xc00d336b10?}, {0x100f71bd0?, 0xc000176a20?}, {0xc00bcd74c0, 0x176a01?, 0x101a5caa0?})
        /Users/user/go/pkg/mod/github.com/klippa-app/go-pdfium@v1.4.1/webassembly/imports/callbacks.go:32 +0xc7
github.com/tetratelabs/wazero/internal/engine/compiler.(*callEngine).execWasmFunction(0xc00bcce500, {0x100f6c788?, 0xc000136748?}, 0xc00bcce500?)
        /Users/user/go/pkg/mod/github.com/tetratelabs/wazero@v1.1.0/internal/engine/compiler/engine.go:1007 +0x1b2
github.com/tetratelabs/wazero/internal/engine/compiler.(*callEngine).call(0xc00bcce500, {0x100f6c788, 0xc000136748}, {0xc000abea20?, 0x10?, 0xc000101b00?}, {0x0, 0x0, 0x0})
        /Users/user/go/pkg/mod/github.com/tetratelabs/wazero@v1.1.0/internal/engine/compiler/engine.go:758 +0x2c5
github.com/tetratelabs/wazero/internal/engine/compiler.(*callEngine).Call(0xc000176a20?, {0x100f6c788?, 0xc000136748?}, {0xc000abea20?, 0x10?, 0x101a5c708?})
        /Users/user/go/pkg/mod/github.com/tetratelabs/wazero@v1.1.0/internal/engine/compiler/engine.go:712 +0xd9
github.com/klippa-app/go-pdfium/internal/implementation_webassembly.(*PdfiumImplementation).OpenDocument(0xc00574c240, 0xc003998690)
        /Users/user/go/pkg/mod/github.com/klippa-app/go-pdfium@v1.4.1/internal/implementation_webassembly/implementation.go:255 +0x753
github.com/klippa-app/go-pdfium/webassembly.(*pdfiumInstance).OpenDocument(0xc00048d9e0?, 0x0?)
        /Users/user/go/pkg/mod/github.com/klippa-app/go-pdfium@v1.4.1/webassembly/generated.go:6031 +0xca

```

```
fatal error: concurrent map writes

goroutine 29 [running]:
github.com/klippa-app/go-pdfium/internal/implementation_webassembly.(*PdfiumImplementation).CreateFileAccessReader(0xc0011e3560, 0x19ccba, {0x100f69370?, 0xc00d29c000})
        /Users/user/go/pkg/mod/github.com/klippa-app/go-pdfium@v1.4.1/internal/implementation_webassembly/data.go:51 +0x210
github.com/klippa-app/go-pdfium/internal/implementation_webassembly.(*PdfiumImplementation).OpenDocument(0xc0011e3560, 0xc00d1d09f0)
        /Users/user/go/pkg/mod/github.com/klippa-app/go-pdfium@v1.4.1/internal/implementation_webassembly/implementation.go:248 +0x696
github.com/klippa-app/go-pdfium/webassembly.(*pdfiumInstance).OpenDocument(0xc00042db00?, 0x0?)
        /Users/user/go/pkg/mod/github.com/klippa-app/go-pdfium@v1.4.1/webassembly/generated.go:6031 +0xca
```

**Comment by jerbob92** [comment]
Thanks for the report! I will fix this later this week.

**Comment by jerbob92** [comment]
Sorry, took a bit longer than expected, but should be fixed in v1.5.0!

**Comment by semvis123** [comment]
No problem, thanks for the fix!
